.github/workflows/ql-for-ql-tests.yml

```diff
@@ -17,6 +17,10 @@ on:
 env:
   CARGO_TERM_COLOR: always
 
+permissions:
+  contents: read
+  security-events: write
```

Comment:
Is `security-events` necessary at all? I don't see it uploading SARIF.

```suggestion
```

Reply:
Without fixing:
* https://github.com/github/codeql-action/pull/2121

Things fail:
https://github.com/check-spelling-sandbox/codeql/actions/runs/7820458947

Using a fixed version of github/codeql-action, it won't fail:
https://github.com/check-spelling-sandbox/codeql/actions/runs/7824516168/job/21347184612

So, I'll be able to remove this once that PR merges (or considered differently, it's already removed locally, but I can't push that change to this branch until that PR merges).